torchdata/dataloader2/utils/worker.py

```diff
@@ -96,6 +97,9 @@ def process_init_fn(
         datapipe = custom_init_fn(datapipe, worker_info)
         assert isinstance(datapipe, (IterDataPipe, MapDataPipe))
 
+    if worker_prefetch_cnt > 0:
+        datapipe = datapipe.prefetch(worker_prefetch_cnt)
```

Comment:
IIUC, this is still applied whether or not the DataPipe is replicable but now it is after transformation done by the init function?